src/extension/background-script/actions/ln/__tests__/request.test.ts

```diff
@@ -0,0 +1,326 @@
+import utils from "~/common/lib/utils";
+import type Connector from "~/extension/background-script/connectors/connector.interface";
+import db from "~/extension/background-script/db";
+import type { MessageGenericRequest, OriginData } from "~/types";
+
+import request from "../request";
+
+// suppress console logs when running tests
+console.error = jest.fn();
+
+jest.mock("~/common/lib/utils", () => ({
+  openPrompt: jest.fn(() => Promise.resolve({ data: {} })),
+}));
+
+// overwrite "connector" in tests later
+let connector: Connector;
+const ConnectorClass = jest.fn().mockImplementation(() => {
+  return connector;
+});
+
+jest.mock("~/extension/background-script/state", () => ({
+  getState: () => ({
+    getConnector: jest.fn(() => Promise.resolve(new ConnectorClass())),
+  }),
+}));
+
+const allowanceInDB = {
+  enabled: true,
+  host: "pro.kollider.xyz",
+  id: 1,
+  imageURL: "https://pro.kollider.xyz/favicon.ico",
+  lastPaymentAt: 0,
+  lnurlAuth: true,
+  name: "pro kollider",
+  remainingBudget: 500,
+  totalBudget: 500,
+  createdAt: "123456",
+  tag: "",
+};
+
+const permissionInDB = {
+  id: 1,
+  allowanceId: allowanceInDB.id,
+  createdAt: "1487076708000",
+  host: allowanceInDB.host,
+  method: "webln/makeinvoice",
+  blocked: false,
+  enabled: true,
+};
+
+const message: MessageGenericRequest = {
+  action: "request",
+  origin: { host: allowanceInDB.host } as OriginData,
+  args: {
+    method: "makeInvoice",
+    params: {},
+  },
+};
+
+const requestResponse = { data: [] };
+const fullConnector = {
+  requestMethod: jest.fn(() => Promise.resolve(requestResponse)),
+  supportedMethods: [
+    // saved and compared in lowercase
+    "getinfo",
+    "makeinvoice",
+    "sendpayment",
+  ],
+} as unknown as Connector;
+
+// prepare DB with allowance
+db.allowances.bulkAdd([allowanceInDB]);
+
+// resets after every test
+afterEach(async () => {
+  jest.clearAllMocks();
+  // ensure a clear permission table in DB
+  await db.permissions.clear();
+  // set a default connector if overwritten in a previous test
+  connector = fullConnector;
+});
+
+describe("ln request", () => {
+  describe("throws error", () => {
+    test("if connector does not support requestMethod", async () => {
+      connector = {
+        ...fullConnector,
+        supportedMethods: ["getinfo"],
+      };
+
+      const result = await request(message);
+
+      expect(console.error).toHaveBeenCalledTimes(1);
+      expect(result).toStrictEqual({
```

Comment:
Can something like this be used to expect an error?
https://github.com/getAlby/lightning-browser-extension/pull/1875/files#diff-54f3405772ff247a2581cbc399cfb41599e415a405be1ba9c4a305e29b14ada6R50-R52

Reply:
I don't think so.
`request` will always return an object in the catch block.
https://github.com/getAlby/lightning-browser-extension/blob/master/src/extension/background-script/actions/ln/request.ts#L86-L92
and not a JavaScript Error